modules/consensus/database.go

```diff
@@ -78,6 +78,29 @@ var (
 	SiafundPool = []byte("SiafundPool")
 )
 
+type (
+	// dbBucket represents a collection of key/value pairs inside the database.
+	dbBucket interface {
+		Get(key []byte) []byte
+	}
+
+	// dbTx represents a read-only transaction on the database that can be used
+	// for retrieving values.
+	dbTx interface {
+		Bucket(name []byte) dbBucket
+	}
+
+	// boltTxWrapper wraps a bolt.Tx so that it matches the dbBucket interface.
+	boltTxWrapper struct {
+		tx *bolt.Tx
+	}
+)
+
+// Bucket returns the dbBucket associated with the given bucket name.
+func (b boltTxWrapper) Bucket(name []byte) dbBucket {
+	return b.tx.Bucket(name)
+}
```

Comment:
Can I ask you to split the bucket changes into a separate PR from the clock changes?

I think that changes to how the database stuff works in the consensus package are both needed but also going to be very big. I'm not sure where to proceed with that, but `dbBucket` and `dbTx` seem like a good start.